_posts/2019-11-13-Upgrading-a-Fairly-Large-Rails-Application-from-4.2-to-5.2.markdown

```diff
@@ -0,0 +1,25 @@
+
+<!--more-->
+
+We spoke with Marcus McLaughlin, Head of Engineering at [Amitree](https://www.home.amitree.com/), about the work performed to upgrade their app. Marcus states that their application is fairly large, about 5 years old and has a heavy reliance on background processing.
+
+[Amitree](https://www.home.amitree.com/) was having difficulty justifying developer cycles to the Rails upgrade project because they wanted to focus on growth as opposed to maintenance, a common issue experienced by many companies. Their application fell out of date because, in Marcus' words, "upgrading didn’t make the cut for our priority list, and we never had a business critical reason to upgrade the application until the version we were running was about to go entirely out of support and could result in security issues". That is why they decided to come to Ombu Labs for help.
+
+Initially, [Amitree](https://www.home.amitree.com/) ordered the Ombu Labs Roadmap so they could have a good estimate of what would be needed to do the upgrade. The roadmap proved to be useful "both in terms of the information that it provided, but also in terms of building my [Marcus'] confidence in Ombu Labs ability to deliver the estimate".
+
+We then executed a full upgrade of the application from Rails 4.2 to 5.2. This necessary upgrade of their application ensured their application continued running on a Rails version that is still supported, preventing security issues and bugs, something that is of critical importance for [Amitree](https://www.home.amitree.com/) because they prioritize the security and safety of user data.
+
+According to Marcus, the Ombu Labs team was responsive and self-sufficient, which resulted in the time the [Amitree](https://www.home.amitree.com/) team needed to dedicate to the upgrade being "considerably less than I [Marcus] would have expected". OmbuLabs' exclusive focus on the Rails upgrade allowed the [Amitree](https://www.home.amitree.com/) team to continue focusing on achieving their other business goals. In the end, they had a fully upgraded Rails application and had "zero major production issues as a result of the upgrade process. During the upgrade, they also saw "very little decrease in productivity as a team during the upgrade."
+
+On a final note, Marcus also states he found the OmbuLabs team "pleasant to work with and knowledgeable". Working with OmbuLabs allowed Amitree to continue to focus on growth while ensuring a high level of security standards for their customers.
+
+For more information about upgrading your Rails application, check out our "[Upgrade Rails Series](https://www.ombulabs.com/blog/tags/upgrades)", a series of do-it-yourself guides to upgrading Rails.
```

Comment:
@ABizzinotto Please make this link point to fastruby.io/blog/tags/upgrades instead.

Reply:
Done :)